docs/faq.rst

```diff
@@ -353,8 +353,52 @@ method for this purpose. For example::
                     for _ in range(item['multiply_by']):
                         yield deepcopy(item)
 
+.. _faq-dataclass-items:
+
+Can I use dataclasses as items?
+-------------------------------
+
+Support for :class:`dataclasses.dataclass` objects as items was added in version 2.0.
+This works natively in Python 3.7+, or using the `dataclasses backport`_ in Python 3.6.
+
+Most of the examples in this documentation assume you are using either :class:`dict`
+or :class:`~scrapy.item.Item` objects, and access their values in a dictionary-like
+manner. However, ``dataclass`` objects expose their values trough attributes instead,
+which means you might need to update your existing components
+(such as :ref:`item pipelines <topics-item-pipeline>`,
+:ref:`signal handlers <topics-signals>` handlers or
+:ref:`spider middlewares <topics-spider-middleware>`) to make them work correctly.
+
+Alternatively, you can use the ``scrapy.utils.decorators.subscriptable_dataclass``
+decorator, which adds the appropriate methods in order to make ``dataclass`` objects
+capable of being accessed like dictionaries:
```

Comment:
I think we should suggest this decorator to make items work with third-party middlewares/signal handlers which are not updated yet to support dataclasses.